File: app/src/main/java/com/example/minilist/SettingsPresenter.java
package com.example.minilist;

import android.content.SharedPreferences;
import android.preference.PreferenceManager;
import android.widget.RadioGroup;

import static com.example.minilist.SettingsActivity.SETTINGS_TEXT_COLOR;
import static com.example.minilist.SettingsActivity.SETTINGS_TEXT_SIZE;

public class SettingsPresenter {

    private final SettingsActivity wireframe;

    SettingsPresenter(SettingsActivity wireframe, SharedPreferences sharedPreferences) {
        this.wireframe = wireframe;
        switch (sharedPreferences.getInt(SETTINGS_TEXT_SIZE, R.dimen.average_text_size)) {
            case R.dimen.little_text_size:
                wireframe.setSizeRadio(R.id.radio_14);
                break;
            case R.dimen.big_text_size:
                wireframe.setSizeRadio(R.id.radio_18);
                break;
            default:
                wireframe.setSizeRadio(R.id.radio_16);
                break;
        }


        switch (sharedPreferences.getInt(SETTINGS_TEXT_COLOR, R.color.black)) {
            case R.color.burnt_umber:
                wireframe.setColorRadio(R.id.radio_burnt_umber);
                break;
            case R.color.kashmir_green:
                wireframe.setColorRadio(R.id.radio_kashmir_green);
                break;
            default:
                wireframe.setColorRadio(R.id.radio_black);
                break;
        }

    }

    public void changeSize(RadioGroup radioGroup) {
        int size;
        switch (radioGroup.getCheckedRadioButtonId()) {
            case R.id.radio_14:
                size = R.dimen.little_text_size;
                break;
            case R.id.radio_18:
                size = R.dimen.big_text_size;
                break;
            default:
                size = R.dimen.average_text_size;
                break;
        }
        SharedPreferences.Editor save = PreferenceManager.getDefaultSharedPreferences(wireframe).edit();
        save.putInt(SETTINGS_TEXT_SIZE, size);
        save.apply();
    }

    public void changeColor(RadioGroup radioGroup) {
        SharedPreferences.Editor save = PreferenceManager.getDefaultSharedPreferences(wireframe).edit();
        int color;
        switch (radioGroup.getCheckedRadioButtonId()) {
            case R.id.radio_burnt_umber:
                color = R.color.burnt_umber;
                break;
            case R.id.radio_kashmir_green:
                color = R.color.kashmir_green;
                break;
            default:
                color = R.color.black;
                break;
        }
        save.putInt(SETTINGS_TEXT_COLOR, color);
        save.apply();
    }
}

File: app/src/main/java/com/example/miniListMVVM/settings/SettingsPresenter.kt
package com.example.miniListMVVM.settings

import android.content.SharedPreferences
import com.example.miniListMVVM.R
import com.example.miniListMVVM.Shared

class SettingsPresenter internal constructor(wireframe: SettingsWireframe, private val sharedPreferences: SharedPreferences) {

    init {
        wireframe.setSizeRadio(when (sharedPreferences.getInt(Shared.SETTINGS_TEXT_SIZE, Shared.SIZE_AVERAGE)) {
            Shared.SIZE_LITTLE -> R.id.radio_little
            Shared.SIZE_BIG -> R.id.radio_big
            else -> R.id.radio_avg
        })
        wireframe.setColorRadio(when (sharedPreferences.getInt(Shared.SETTINGS_TEXT_COLOR, Shared.COLOR_BLACK)) {
            Shared.COLOR_UMBER -> R.id.radio_burnt_umber
            Shared.COLOR_GREEN -> R.id.radio_kashmir_green
            else -> R.id.radio_black
        })
    }

    fun changeSize(checkedRadioButtonId: Int) {
        sharedPreferences.edit()
                .putInt(
                        Shared.SETTINGS_TEXT_SIZE,
                        when (checkedRadioButtonId) {
                            R.id.radio_little -> Shared.SIZE_LITTLE
                            R.id.radio_big -> Shared.SIZE_BIG
                            else -> Shared.SIZE_AVERAGE
                        }
                )
                .apply()
    }

    fun changeColor(checkedRadioButtonId: Int) {
        sharedPreferences.edit().putInt(Shared.SETTINGS_TEXT_COLOR, when (checkedRadioButtonId) {
            R.id.radio_burnt_umber -> Shared.COLOR_UMBER
            R.id.radio_kashmir_green -> Shared.COLOR_GREEN
            else -> Shared.COLOR_BLACK
        }).apply()
    }
}